File: python_solution/problem28.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
"""
Created on Tue Sep 24 20:46:48 2024

@author: sharminalina
"""

class Problem28:
    def strStr(self, haystack: str, needle: str) -> int:
        if needle == "":
            return 0
        
        j = len(needle)

        for i in range(len(haystack)):
            if haystack[i:i+j] == needle:
                return i
        
        return -1
            
                
    
def main():
    print("Solution of Leet problem No.28: Find the index of first occurance in a string")
    
    ## Taking input first string from the console
    haystack = input("Enter the first string: ")
    needle = input("Enter the first string: ")
    
    solution_prob28 = Problem28()
    result = solution_prob28.strStr(haystack, needle)
    
    # Printing the index of first occurance in the given string
    print(f"Length after removing the elements: {result}")
    

    
if __name__ == "__main__":
    main()

File: C#Solutions/problem28.cs
using System;
using System.Collections.Generic;

namespace AllProblems
{
    public static class Problem28
    {
        public static void Solve()
        {
            Console.WriteLine("Solution to Problem 28");
            // Read the array from console
            Console.WriteLine("Enter a string:");
            string haystack= Console.ReadLine();
            Console.WriteLine("Enter the 2nd string:");
            string needle= Console.ReadLine();

    
            // Find the Find the index of first occurance in a string
            int result = strStr(haystack,needle);
            Console.WriteLine($"the index of first occurance in a string: {result}");
        }

        

        public static int strStr(string haystack, string needle) {
            if(needle.Length == 0){
                return 0;
            }
            for (int i = 0; i < haystack.Length - needle.Length; i++){
                if (haystack.Substring(i, needle.Length) == needle){
                    return i;
                }
            
            }
            return -1;


            
        }
    }
}